engine/filters.go

```diff
@@ -357,16 +331,16 @@ func (fltr *FilterRule) Pass(dP config.DataProvider,
 	return result != *(fltr.negative), nil
 }
 
-func (fltr *FilterRule) passString(dP config.DataProvider) (bool, error) {
-	strVal, err := config.GetDynamicString(fltr.FieldName, dP)
+func (fltr *FilterRule) passString(fielNameDP config.DataProvider, fieldValuesDP []config.DataProvider) (bool, error) {
+	strVal, err := config.GetDynamicString(fltr.FieldName, fielNameDP)
```

Comment:
Please rename the GetDynamicString function (and interface) to DPDynamicString and DPDynamicInterface()